vortex-dtype/src/ptype.rs

```diff
@@ -150,6 +151,32 @@ impl PType {
     }
 }
 
+impl DType {
+    pub fn is_unsigned_int(&self) -> bool {
+        PType::try_from(self)
+            .map(|ptype| ptype.is_unsigned_int())
+            .unwrap_or_default()
```

Comment:
nit: personally I dislike `unwrap_or_default`, since it requires the reader to think/know the default of the type. For types that a trivial to instantiate, the explicit version is generally clearer -> i.e., `unwrap_or(false)`

Reply:
I think clippy yells at you for that. Maybe we can configure it to be opposite for primitives?

Reply:
You're not the only one: https://github.com/rust-lang/rust-clippy/discussions/11630